src/rust/engine/src/nodes.rs

```diff
@@ -271,6 +271,24 @@ impl WrappedNode for Select {
             })
             .to_boxed()
         }
+        &rule_graph::Rule::Intrinsic(Intrinsic { product, input })
+          if product == context.core.types.snapshot
+            && input == context.core.types.files_content =>
+        {
+          let core = context.core.clone();
+          self
+            .select_product(&context, context.core.types.files_content, "intrinsic")
+            .and_then(move |files_content| {
+              let files_content: Vec<_> = externs::project_multi(&files_content, "dependencies")
+                .iter()
+                .map(lift_file_content)
+                .collect();
+              fs::Snapshot::from_files_content(context.core.store(), files_content)
+                .map_err(|str| throw(&str))
```

Comment:
`str` gets syntax highlighted as a type, so might be good to avoid using it as a variable name.

Reply:
Renamed